src/cli/cmd/generate/generate.go

```diff
@@ -197,3 +205,69 @@ func copyLines(srcDir, destDir, relFile string, ranges ...[2]int) error {
 
 	return writer.Flush()
 }
+
+func copyProfileConfigurationFile(moduleDirPath string, profileName string, codeDestDirPath string) error {
+	sourcePath, err := getProfileVariableInputSourceFile(moduleDirPath, profileName)
+	if err != nil {
+		return fmt.Errorf("could not retrieve configuration file for platform profile '%s': %w", profileName, err)
+	}
+
+	destPath, err := getProfileVariableInputDestFile(codeDestDirPath)
+	if err != nil {
+		return fmt.Errorf("could not retrieve configuration target path for platform profile '%s': %w", profileName, err)
+	}
+
+	if copyFile(sourcePath, destPath); err != nil {
+		return fmt.Errorf("could not copy platform '%s' profile configuration: %w", profileName, err)
+	}
+
+	return nil
+}
+
+func getProfileVariableInputSourceFile(moduleDirPath string, profileName string) (filePath string, err error) {
+	profileSourceFileName := fmt.Sprintf("%s.%s", profileName, "tfvars")
+	profileSourceFilePath := path.Join(moduleDirPath, profileSourceFileName)
+
+	sourceFileStat, err := os.Stat(profileSourceFilePath)
+	if os.IsNotExist(err) {
+		return profileSourceFilePath, fmt.Errorf("platform must define configuration for profile '%s' in terraform input file '%s'", profileName, profileSourceFilePath)
```

Comment:
We should use eris library for creating & wrapping error objects. It records error stack as well. In unit test framework, I have it integrated such that in case of unexpected errors, we get the stack of where the error came from. And in future we can add a debug log of stack trace in all errors as well to further debug the origin of the error.

Reply:
Ok. I am going to merge this and wrap all errors in a separate patch.